pkg/common/commands/resetoffset/controller/dataplane.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "knative.dev/control-protocol/pkg"
+	ctrlmessage "knative.dev/control-protocol/pkg/message"
+	ctrlreconciler "knative.dev/control-protocol/pkg/reconciler"
+	"knative.dev/pkg/logging"
+
+	kafkav1alpha1 "knative.dev/eventing-kafka/pkg/apis/kafka/v1alpha1"
+	"knative.dev/eventing-kafka/pkg/common/commands/resetoffset/refmappers"
+	"knative.dev/eventing-kafka/pkg/common/controlprotocol/commands"
+)
+
+const (
+	asyncCommandResultPollDuration    = 1 * time.Second  // AsyncCommandResult Polling Duration
+	asyncCommandResultTimeoutDuration = 10 * time.Second // AsyncCommandResult Timeout Duration
+
+	ControlProtocolServerPort = 8085 // TODO - Use common constant once eric delivers!
+)
+
+// reconcileDataPlaneServices updates the Reconciler ConnectionPool Services associated with the specified RefInfo.
+func (r *Reconciler) reconcileDataPlaneServices(ctx context.Context, resetOffset *kafkav1alpha1.ResetOffset, refInfo *refmappers.RefInfo) (map[string]ctrl.Service, error) {
+
+	// Get The Logger From Context
+	logger := logging.FromContext(ctx).Desugar().With(zap.Any("RefInfo", refInfo))
+
+	// Create A Control-Protocol PodIpGetter & Get The Pod IPs
+	podIpGetter := ctrlreconciler.PodIpGetter{Lister: r.podLister}
+	podIPs, err := podIpGetter.GetAllPodsIp(refInfo.DataPlaneNamespace, labels.Set(refInfo.DataPlaneLabels).AsSelector())
+	if err != nil {
+		logger.Error("Failed to getting data-plane Pod IPs", zap.Error(err))
+		return nil, err
+	}
+
+	// Append The Control-Protocol Server Port Number To The PodIPs If Not Already Present
+	for index, podIP := range podIPs {
+		if !strings.Contains(podIP, ":") {
+			podIPs[index] = fmt.Sprintf("%s:%d", podIP, ControlProtocolServerPort)
+		}
+	}
```

Comment:
I don't think this is necessary if you use the default control protocol port, it should automatically append the port if no one is specified

Reply:
In #724 @eric-sap has created a constant (temporarily mirrored in this PR) for the Port number instead of relying on the control-protocol default.  Therefore, I am coding against the "same" constant here for consistency.  I don't care too much whether we use the default, a constant, or expose it as a config though - whatever you think is best, We can change both the PRs either way.